src/dist/dist.rs

```diff
@@ -406,16 +406,12 @@ impl TargetTriple {
 impl std::convert::TryFrom<PartialTargetTriple> for TargetTriple {
     type Error = &'static str;
     fn try_from(value: PartialTargetTriple) -> std::result::Result<Self, Self::Error> {
-        if value.arch.is_some() && value.os.is_some() && value.env.is_some() {
-            Ok(Self(format!(
-                "{}-{}-{}",
-                value.arch.unwrap(),
-                value.os.unwrap(),
-                value.env.unwrap()
-            )))
-        } else {
-            Err("Incomplete / bad target triple")
-        }
+        let arch = value.arch.ok_or("Incomplete / bad target triple")?;
+        let os = value.os.ok_or("Incomplete / bad target triple")?;
```

Comment:
Sorry for the late review 🙇, but wouldn't it be problematic to pass `arch` and `os` as strings as if they were valid? Is there a particular reason not to fail early here?

Reply:
I don't understand. The `ok_or` transforms the `Option<T>` into a `Result<T, E>`, so, these two lines would be fail early if any of `arch` or `os` is None.

Reply:
@ongchi Thanks for your explanation! Must have been a misread from my side, sorry 🙇